.github/actions/action.yml

```
@@ -0,0 +1,31 @@
+name: 'Run Tests'
+description: 'Run Tests'
+inputs:
+  test_selection:
+    description: 'Positional Argument for stestr run'
+    default: ""
+    required: false
+runs:
+  using: "composite"
+  steps:
+    - name: Clone mtreinish/qiskit-neko PUBLIC repository
+      uses: mtreinish/clone-github-repo-action@v2
+      with:
+        owner: 'mtreinish'
+        repository: 'qiskit-neko'
+    - name: Access cloned repository content
+      run: |
+        cd qiskit-neko
+    - name: Download Released Dependencies
+      run: |
+        pip install .
+        cd ..
+    - name: Checkout base repository
+      uses: actions/checkout@v3   
```

I think we still probably need a `pip install` here after doing the checkout. As some of the dependencies (mainly aer and terra) have compiled extensions and we need to support building them. Something like:

```suggestion
    - name: Checkout base repository
      uses: actions/checkout@v3
    - name: Install repo under test
      run: pip install .  
```

(assuming after running `actions/checkout` the working directory is the repo root)

Although thinking about it some more I think having this new install repo under test step configurable would be good. So maybe also add an option like:

```
  repo_install_command:
    description: 'Optional command to run for installing repository under test'
    default: "pip install ."
    required: false
```
and then use `${{ inputs.repo_install_command }}` for the `run:` field on the new step.

---

Yes! Adding a separate input to install the repo under test makes sense. I have added the input variable and have made the changes.